src/plugins/postgres_check_operator.py

```diff
@@ -1,139 +1,305 @@
-    greater than yesterday's partition, or that a set of metrics are less
-    than 3 standard deviation for the 7 day average.
-
-    This operator can be used as a data quality check in your pipeline, and
-    depending on where you put it in your DAG, you have the choice to
-    stop the critical path, preventing from
-    publishing dubious data, or on the side and receive email alerts
-    without stopping the progress of the DAG.
+    SELECT column_name FROM information_schema.columns
+     WHERE table_schema = 'public' AND table_name = %s
+     ORDER BY column_name
+""",
+    result_fetcher=flattened_records_as_set,
+)
 
-    :param sql: the sql to be executed
-    :type sql: str
-    :param postgres_conn_id: reference to the Postgres database
-    :type postgres_conn_id: str
+GEO_CHECK = CheckFactory(
+    """
+  {% set lparams = params['$check_id'] %}
+  {% set geo_column = lparams.geo_column|default("geometry", true) %}
+  SELECT 1 WHERE NOT EXISTS (
+      SELECT FROM {{ lparams.table_name }} WHERE
+        {{ geo_column }} IS null
+        {% if lparams.check_valid|default(true) %} OR ST_IsValid({{ geo_column }}) = false {% endif %}
+          OR GeometryType({{ geo_column }})
+        {% if lparams.geotype is string %}
+          <> '{{ lparams.geotype }}'
+        {% else %}
+          NOT IN ({{ lparams.geotype | map('quote') | join(", ") }})
+        {% endif %}
+      )
+""",
+    result_fetcher=first_flattened_record,
+)
+
+
+class PostgresMultiCheckOperator(BaseOperator):
+    """ This operator can be used to fire a number of checks at once.
+        This is more efficient than firing up all checks separately.
+        There is one caveat. Because we want to use the efficient and
+        lazily evaluated jinja templating, the template parameters have
+        to be collected into one dict, because Airflow does the parameter
+        interpolation only once. So, when params are used, they have to
+        be collected with the 'make_params' function and fed into the
+        operator constructor.
     """
 
-    template_fields = ("sql",)
-    template_ext = (".sql",)
+    # We use the possibilty to have nested template fields here
+    template_fields: Iterable[str] = ["checks"]
 
     @apply_defaults
-    def __init__(self, sql, postgres_conn_id="postgres_default", *args, **kwargs):
-        super(PostgresCheckOperator, self).__init__(sql=sql, *args, **kwargs)
+    def __init__(
+        self, postgres_conn_id="postgres_default", checks=[], *args, **kwargs,
+    ):
+        super(PostgresMultiCheckOperator, self).__init__(*args, **kwargs)
         self.postgres_conn_id = postgres_conn_id
-        self.sql = sql
+        self.checks = checks
+        self.check_checks_for_params(kwargs.get("params"))
+
+    def check_checks_for_params(self, params):
+        if not params and reduce(
+            lambda acc, d: {**acc, **(d or {})},
+            list(make_params(self.checks).values()),
+            {},
+        ):
```

Comment:
Wat gebeurt hier precies?

Komt dat overeen met dit?
```suggestion
        if not params and any(acc or d for acc, d in make_params(self.checks).values()):
```